videoprocess/vpphdr_tm.cpp

```diff
@@ -0,0 +1,861 @@
+        if (strncmp(field, field_name, strlen(field_name)))
+            continue;
+
+        if (!(str = strtok(NULL, ":")))
+            continue;
+
+        /* skip blank space in string */
+        while (*str == ' ')
+            str++;
+
+        *(str + strlen(str)-1) = '\0';
+        strcpy(value, str);
+
+        return 0;
+    }
+    return -1;
+}
+
+static int8_t
+read_value_uint32(FILE* fp, const char* field_name, uint32_t* value)
+{
+    char str[MAX_LEN];
+
+    if (read_value_string(fp, field_name, str)) {
+       printf("Failed to find integer field: %s", field_name);
+       return -1;
+    }
+
+    *value = (uint32_t)atoi(str);
+
+    return 0;
+}
+
+static int8_t
+read_value_float(FILE *fp, const char* field_name, float* value)
+{
+    char str[MAX_LEN];
+    if (read_value_string(fp, field_name, str)) {
+       printf("Failed to find float field: %s \n",field_name);
+       return -1;
+    }
+
+    *value = atof(str);
+
+    return 0;
+}
+
+static VAStatus
+create_surface(VASurfaceID * p_surface_id,
+               uint32_t width, uint32_t height,
+               uint32_t fourCC, uint32_t format)
+{
+    VAStatus va_status;
+    VASurfaceAttrib    surface_attrib;
+    surface_attrib.type =  VASurfaceAttribPixelFormat;
+    surface_attrib.flags = VA_SURFACE_ATTRIB_SETTABLE;
+    surface_attrib.value.type = VAGenericValueTypeInteger;
+    surface_attrib.value.value.i = fourCC;
+
+    va_status = vaCreateSurfaces(va_dpy,
+                                 format,
+                                 width ,
+                                 height,
+                                 p_surface_id,
+                                 1,
+                                 &surface_attrib,
+                                 1);
+
+    printf("create_surface: p_surface_id %d, width %d, height %d, fourCC 0x%x, format 0x%x\n", 
+        *p_surface_id, width, height, fourCC, format);
+
+    return va_status;
+}
+
+static VAStatus
+hdrtm_filter_init(VABufferID *filter_param_buf_id, uint32_t tm_type)
+{
+    VAStatus va_status = VA_STATUS_SUCCESS;
+    VAProcFilterParameterBufferHDRToneMapping hdrtm_param;
+
+    VAHdrMetaDataHDR10 in_hdr10_metadata = {};
+
+	// The input is HDR content
+    in_hdr10_metadata.max_display_mastering_luminance = g_in_max_display_luminance;
+    in_hdr10_metadata.min_display_mastering_luminance = g_in_min_display_luminance;
+    in_hdr10_metadata.max_content_light_level         = g_in_max_content_luminance;
+    in_hdr10_metadata.max_pic_average_light_level     = g_in_pic_average_luminance;
+    in_hdr10_metadata.display_primaries_x[0] = 8500;
+    in_hdr10_metadata.display_primaries_y[0] = 39850;
+    in_hdr10_metadata.display_primaries_x[1] = 35400;
+    in_hdr10_metadata.display_primaries_y[1] = 14600;
+    in_hdr10_metadata.display_primaries_x[2] = 6550;
+    in_hdr10_metadata.display_primaries_y[2] = 2300;
+    in_hdr10_metadata.white_point_x = 15635;
+    in_hdr10_metadata.white_point_y = 16450; 
```

Comment:
these values is standard meta data?

Reply:
Yes, these value is BT2020 standard color primaries.